src/main/java/com/iexec/core/sms/SmsService.java

```diff
@@ -119,27 +117,25 @@ public Optional<String> getEnclaveChallenge(String chainTaskId, String smsUrl) {
                 : generateEnclaveChallenge(chainTaskId, smsUrl);
     }
 
-    @Retryable(value = FeignException.class)
     Optional<String> generateEnclaveChallenge(String chainTaskId, String smsUrl) {
         // SMS client should already have been created once before.
         // If it couldn't be created, then the task would have been aborted.
         // So the following won't throw an exception.
         final SmsClient smsClient = smsClientProvider.getSmsClient(smsUrl);
 
-        final String teeChallengePublicKey = smsClient.generateTeeChallenge(chainTaskId);
+        try {
+            final String teeChallengePublicKey = smsClient.generateTeeChallenge(chainTaskId);
+
+            if (StringUtils.isEmpty(teeChallengePublicKey)) {
+                log.error("An error occurred while getting teeChallengePublicKey "
+                        + "[chainTaskId:{}, smsUrl:{}]", chainTaskId, smsUrl);
+                return Optional.empty();
+            }
 
-        if (StringUtils.isEmpty(teeChallengePublicKey)) {
-            log.error("An error occurred while getting teeChallengePublicKey " 
-                + "[chainTaskId:{}, smsUrl:{}]", chainTaskId, smsUrl);
+            return Optional.of(teeChallengePublicKey);
+        } catch (FeignException e) {
```

Comment:
With a reverse proxy, other runtime exceptions could happen in production.
Do we want to have the biggest net and catch all exceptions here ?

Reply:
Thanks for the idea. Here is a commit addressing the issue: https://github.com/iExecBlockchainComputing/iexec-core/pull/663/commits/a8997d4d9aca42f3c13989da2da320a2f9f2b8a6
Catching unchecked exceptions should be enough there, as Feign methods contract do not allow them to throw checked exceptions.